source/MaterialXGenGlsl/GlslResourceBindingContext.cpp

```diff
@@ -0,0 +1,45 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenGlsl/GlslResourceBindingContext.h>
+
+namespace MaterialX
+{
+
+//
+// GlslResourceBindingContext
+//
+GlslResourceBindingContext::GlslResourceBindingContext()
+{
+    _requiredExtensions.insert("GL_ARB_shading_language_420pack");
+}
+
+void GlslResourceBindingContext::emitUniformBlock(GenContext& context, const VariableBlock& uniforms, SyntaxPtr syntax, ShaderStage& stage)
```

Comment:
Do we need to worry about ubo sizes for uniforms and sampler blocks? Might be good to do a max size check
and output multiple blocks if needed?

Reply:
The size is implementation dependent. It's similar to saying in glsl 400 we emit N uniforms that are internally part of a default uniform block. With GL_ARB_shading_language_420pack, we are just group them.

Reply:
in other words, if we generate a glsl shader with N uniforms where N > max supported uniforms, the shader will fail to compile.